File: Kattis/countingstars.py
import resource, sys
sys.setrecursionlimit( 10**6 )

def floodfill( y, x, number ):

	if table[y][x] != '-':
		return
	table[y][x] = number

	if y + 1 < m:
		floodfill( y + 1, x, number )
	if y > 0:
		floodfill( y - 1, x, number )
	if x + 1 < n:
		floodfill( y, x + 1, number )
	if x > 0:
		floodfill( y, x - 1, number )
	return

counter = 0

while True:

	counter += 1

	try:
		m, n  = input().split()
		m = int( m )
		n = int( n )
	except EOFError:
		break

	table = []

	found = 0

	for _ in range( m ):

		table.append( list( input() ) )

	for y in range( m ):
		for x in range( n ):

			if table[y][x] == '-':
				floodfill( y, x, found )
				found += 1

	print( "Case %d: %d" %( counter, found ) )

File: Kattis/countingstars.cpp
#include <iostream>

using namespace std;

void floodfill( char table[100][100], int x, int y, int m, int n, int c ) {

	table[x][y] = '#';

	if ( x > 0 && table[x - 1][y] == '-' )
		floodfill( table, x - 1, y, m, n, c );
	if ( y > 0 && table[x][y - 1] == '-' )
		floodfill( table, x, y - 1, m, n, c );
	if ( x < m - 1 && table[x + 1][y] == '-' )
		floodfill( table, x + 1, y, m, n, c );
	if ( y < n - 1 && table[x][y + 1] == '-' )
		floodfill( table, x, y + 1, m, n, c );
}

int main() {

	int  m, n, counter, loop = 1;
	char table[100][100];

	while ( cin >> m >> n ) {

		counter = 0;

		for ( int i = 0; i < m; i++ )
			cin >> table[i];

		for ( int i = 0; i < m; i++ )
			for ( int j = 0; j < n; j++ )
				if ( table[i][j] == '-' )
					floodfill( table, i, j, m, n, counter++ );

		cout << "Case " << loop++ << ": " << counter << endl;;
	}

	return 0;
}
